### PR TITLE
[grpc][Gpr_To_Absl_Logging] Migrating from gpr to absl logging - gpr_log

### PR DESCRIPTION
**Pull request by tanvi-jagtap**
[grpc][Gpr_To_Absl_Logging] Migrating from gpr to absl logging - gpr_log
In this CL we are migrating from gRPCs own gpr logging mechanism to absl logging mechanism. The intention is to deprecate gpr_log in the future.

We have the following mapping

1. gpr_log(GPR_INFO,...) -> LOG(INFO)
2. gpr_log(GPR_ERROR,...) -> LOG(ERROR)
3. gpr_log(GPR_DEBUG,...) -> VLOG(2)

Reviewers need to check :

1. If the above mapping is correct.
2. The content of the log is as before.
gpr_log format strings did not use string_view or std::string . absl LOG accepts these. So there will be some elimination of string_view and std::string related conversions. This is expected.

### CODE REVIEW

File: test/core/security/credentials_test.cc [hunk 1]
```diff
@@ -1028,7 +1025,7 @@ void validate_sts_token_http_request(const grpc_http_request* request,
       absl::StrFormat("%s?%s", test_sts_endpoint_url, body);
   absl::StatusOr<URI> url = URI::Parse(get_url_equivalent);
   if (!url.ok()) {
-    gpr_log(GPR_ERROR, "%s", url.status().ToString().c_str());
+    LOG(ERROR) << url.status().ToString();
```
Comment (yousukseung) [inline test/core/security/credentials_test.cc:1028]: I think we should look into using Abseil Stringfy with gRPC types if you have not.

File: test/core/test_util/cmdline_test.cc [hunk 2]
```diff
@@ -20,14 +20,13 @@
...
+#include "absl/log/log.h"
 #include "gtest/gtest.h"
 
-#include <grpc/support/log.h>
-
 #include "src/core/util/useful.h"
 #include "test/core/test_util/test_config.h"
 
-#define LOG_TEST() gpr_log(GPR_INFO, "test at %s:%d", __FILE__, __LINE__)
+#define LOG_TEST() LOG(INFO) << "test at " << __FILE__ << ":" << __LINE__
```
Comment (yousukseung) [inline test/core/test_util/cmdline_test.cc:29]: Maybe unnecessary if abseil logging already adds file/line?

**Review by yousukseung — APPROVED** [review]

File: test/core/test_util/cmdline_test.cc (hunk 2 above)
Comment (tanvi-jagtap) [reply test/core/test_util/cmdline_test.cc:29]: Even gpr_log added file and line . So behaviourally gpr_log and LOG are equivalent. I don't know why they did this.

File: test/core/security/credentials_test.cc (hunk 1 above)
Comment (tanvi-jagtap) [reply test/core/security/credentials_test.cc]: Missed this. Fixed it.

**Review by yousukseung — APPROVED** [review]

**Pull request closed without merging**